File: dotnet/azure/Mcma.Azure.Functions.Api/QueueWorkerInvoker.cs
using System;
using System.Threading.Tasks;
using Mcma.Api;
using Mcma.Core.Context;
using Mcma.Core.Serialization;
using Microsoft.Azure.Storage;
using Microsoft.Azure.Storage.Queue;

namespace Mcma.Azure.Functions.Api
{
    public class QueueWorkerInvoker : WorkerInvoker
    {
        public QueueWorkerInvoker(IContextVariableProvider contextVariableProvider)
            : base(contextVariableProvider)
        {
        }

        protected override async Task InvokeAsync(string workerFunctionId, WorkerRequest request)
        {
            var appStorageConnectionString = ContextVariableProvider.GetRequiredContextVariable("WEBSITE_CONTENTAZUREFILECONNECTIONSTRING");
            if (!CloudStorageAccount.TryParse(appStorageConnectionString, out var appStorageAccount))
                throw new Exception($"Failed to parse app storage connection string '{appStorageConnectionString}'.");

            await appStorageAccount.CreateCloudQueueClient()
                .GetQueueReference(workerFunctionId)
                .AddMessageAsync(new CloudQueueMessage(request.ToMcmaJson().ToString()));
        }
    }
}

File: azure/api/lib/queue-worker-invoker.ts
import { WorkerInvoker, WorkerRequestProperties } from "@mcma/api";
import { QueueServiceClient } from "@azure/storage-queue";

const appStorageConnectionString = process.env.WEBSITE_CONTENTAZUREFILECONNECTIONSTRING;

export async function invokeQueueTriggeredWorker(workerFunctionId: string, workerRequest: WorkerRequestProperties): Promise<void> {
    const queueStorageClient = QueueServiceClient.fromConnectionString(appStorageConnectionString);
    const queueClient = queueStorageClient.getQueueClient(workerFunctionId);
    await queueClient.sendMessage(JSON.stringify(workerRequest));
}

export class QueueWorkerInvoker extends WorkerInvoker {
    constructor() {
        super(invokeQueueTriggeredWorker);
    }
}